### ISSUE TITLE
E2E false positive

### ISSUE THREAD

**Issue by lebsral** [angular/angular#11538]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior** 

False positive on base e2e test
**Expected behavior**

test should fail
**Reproduction of the problem**

**What is the motivation / use case for changing the behavior?**

wrong tests should fail
**Please tell us about your environment:**

OSX 10.11.6
atom
npm
cli server

* **Angular version:** 2.0.0-rc.X

2.0.0-rc.6
* **Browser:** [ Chrome 52  ] 

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
 TS 2.0
* **Node (for AoT issues):** `node --version` =   v6.5.0

changed app.e2e-spec.ts to:
```
import { CatalogPage } from './app.po';

describe('catalog App', function() {
  let page: CatalogPage;

  beforeEach(() => {
    page = new CatalogPage();
  });

  it('should fail, but does not', () => {
    page.navigateTo();
    expect(page.getParagraphText()).toEqual('Some text that does not exist in my app!');
  });
});
```

then I get the following, unexpected pass of the test
```
$ ng e2e
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.

> catalog@0.0.0 pree2e /Users/larsb/Documents/mycode/catalog
> webdriver-manager update

[11:10:58] I/update - chromedriver: file exists /Users/larsb/Documents/mycode/catalog/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.22mac32.zip
[11:10:58] I/update - chromedriver: unzipping chromedriver_2.22mac32.zip
[11:10:59] I/update - chromedriver: setting permissions to 0755 for /Users/larsb/Documents/mycode/catalog/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.22
[11:10:59] I/update - chromedriver: v2.22 up to date
[11:10:59] I/update - selenium standalone: file exists /Users/larsb/Documents/mycode/catalog/node_modules/protractor/node_modules/webdriver-manager/selenium/selenium-server-standalone-2.53.1.jar
[11:10:59] I/update - selenium standalone: v2.53.1 up to date

> catalog@0.0.0 e2e /Users/larsb/Documents/mycode/catalog
> protractor "./protractor.conf.js"

[11:10:59] W/launcher - You have specified both capabilities and multiCapabilities. This will result in capabilities being ignored
[11:10:59] I/direct - Using ChromeDriver directly...
[11:10:59] I/launcher - Running 1 instances of WebDriver
Spec started

  catalog App
    ✓ should fail, but does not

Executed 1 of 1 spec SUCCESS in 0.018 sec.
[11:11:00] I/launcher - 0 instance(s) of WebDriver still running
[11:11:00] I/launcher - chrome #01 passed

All end-to-end tests pass.
```

**Comment by vicb** [comment]
Could you provide a small repro ? (GH repo)

It could be an env setup pb only, hard to say without seeing it.

**Comment by lebsral** [comment]
@vicb 
Here is a sample app that does what I was talking about
https://github.com/lebsral/ng-sample

Here are the commands I used to create it, and then perform the tests that failed...

```
larsb-concourse:mycode larsb$ node --version
v6.5.0
larsb-concourse:mycode larsb$ npm -v
3.10.8
larsb-concourse:mycode larsb$ ng -v
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
angular-cli: 1.0.0-beta.11-webpack.8
node: 6.5.0
os: darwin x64
larsb-concourse:mycode larsb$ ng new sample
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
installing ng2
  create .editorconfig
  create README.md
  create src/app/app.component.css
  create src/app/app.component.html
  create src/app/app.component.spec.ts
  create src/app/app.component.ts
  create src/app/app.module.ts
  create src/app/index.ts
  create src/app/shared/index.ts
  create src/assets/.gitkeep
  create src/assets/.npmignore
  create src/environments/environment.dev.ts
  create src/environments/environment.prod.ts
  create src/environments/environment.ts
  create src/favicon.ico
  create src/index.html
  create src/main.ts
  create src/polyfills.ts
  create src/styles.css
  create src/test.ts
  create src/tsconfig.json
  create src/typings.d.ts
  create angular-cli.json
  create e2e/app.e2e-spec.ts
  create e2e/app.po.ts
  create e2e/tsconfig.json
  create .gitignore
  create karma.conf.js
  create package.json
  create protractor.conf.js
  create tslint.json
Successfully initialized git.
⠹ Installing packages for tooling via npm[execsync v1.0.2] Attempting to compile native extensions.
⠼ Installing packages for tooling via npm[execSync v1.0.2]
    Native code compile failed!!
⠼ Installing packages for tooling via npmStart downloading binary at https://github.com/sass/node-sass/releases/download/v3.9.3/darwin-x64-48_binding.node
⠸ Installing packages for tooling via npm⸨░░░░░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠼ Installing packages for tooling via npm⸨ ░░░░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠴ Installing packages for tooling via npm⸨ ░░░░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠦ Installing packages for tooling via npm⸨  ░░░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠧ Installing packages for tooling via npm⸨   ░░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠇ Installing packages for tooling via npm⸨    ░░░░░░░░░░░░░░⸩ ⠋ https://github.c⠏ Installing packages for tooling via npm⸨      ░░░░░░░░░░░░⸩ ⠋ https://github.c⠋ Installing packages for tooling via npm⸨       ░░░░░░░░░░░⸩ ⠋ https://github.c⠙ Installing packages for tooling via npm⸨          ░░░░░░░░⸩ ⠋ https://github.c⠹ Installing packages for tooling via npm⸨            ░░░░░░⸩ ⠋ https://github.c⠸ Installing packages for tooling via npm⸨              ░░░░⸩ ⠋ https://github.c⠼ Installing packages for tooling via npm⸨                ░░⸩ ⠋ https://github.c⠴ Installing packages for tooling via npm⸨                  ⸩ ⠋ https://github.cBinary downloaded and installed at /Users/larsb/Documents/mycode/sample/node_modules/node-sass/vendor/darwin-x64-48/binding.node
⠹ Installing packages for tooling via npmConsidering PhantomJS found at /usr/local/bin/phantomjs
Looks like an `npm install -g`
Found PhantomJS at /usr/local/lib/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs ...verifying
⠸ Installing packages for tooling via npmWriting location.js file
PhantomJS linked at /usr/local/lib/node_modules/phantomjs-prebuilt/lib/location
⠦ Installing packages for tooling via npm"/Users/larsb/Documents/mycode/sample/node_modules/node-sass/vendor/darwin-x64-48/binding.node" exists. 
 testing binary.
Binary is fine; exiting.
Installed packages for tooling via npm.
larsb-concourse:mycode larsb$ cd sample
larsb-concourse:sample larsb$ npm install
npm WARN deprecated graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.

> fsevents@1.0.14 install /Users/larsb/Documents/mycode/sample/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/larsb/Documents/mycode/sample/node_modules/fsevents/lib/binding/Release/node-v48-darwin-x64/fse.node" is installed via remote

> node-zopfli@1.4.0 install /Users/larsb/Documents/mycode/sample/node_modules/node-zopfli
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download: https://node-zopfli.s3.amazonaws.com/Release/zopfli-v1.4.0-node-v48-darwin-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for node-zopfli@1.4.0 and node@6.5.0 (node-v48 ABI) (falling back to source compile with node-gyp) 
  CXX(target) Release/obj.target/zopfli/src/zopfli-binding.o
  CXX(target) Release/obj.target/zopfli/src/png/zopflipng.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/blocksplitter.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/cache.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/deflate.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/gzip_container.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/hash.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/katajainen.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/lz77.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/squeeze.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/tree.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/util.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/zlib_container.o
  CC(target) Release/obj.target/zopfli/zopfli/src/zopfli/zopfli_lib.o
  CXX(target) Release/obj.target/zopfli/zopfli/src/zopflipng/zopflipng_lib.o
  CXX(target) Release/obj.target/zopfli/zopfli/src/zopflipng/lodepng/lodepng.o
  CXX(target) Release/obj.target/zopfli/zopfli/src/zopflipng/lodepng/lodepng_util.o
  SOLINK_MODULE(target) Release/zopfli.node
  COPY /Users/larsb/Documents/mycode/sample/node_modules/node-zopfli/lib/binding/node-v48-darwin-x64/zopfli.node
  TOUCH Release/obj.target/action_after_build.stamp
sample@0.0.0 /Users/larsb/Documents/mycode/sample
├─┬ angular-cli@1.0.0-beta.11-webpack.8
│ ├── UNMET PEER DEPENDENCY @angular/platform-server@^2.0.0-rc.5
│ ├─┬ angular2-template-loader@0.5.0
│ │ └─┬ codecov@1.0.1
│ │   └─┬ execSync@1.0.2
│ │     └─┬ temp@0.5.1
│ │       └─┬ rimraf@2.1.4
│ │         └── graceful-fs@1.2.3 
│ ├─┬ compression-webpack-plugin@0.3.1
│ │ └─┬ node-zopfli@1.4.0 
│ │   ├── defaults@1.0.3 
│ │   └─┬ node-pre-gyp@0.6.30 
│ │     ├── rc@1.1.6 
│ │     └─┬ tar-pack@3.1.4 
│ │       ├── fstream-ignore@1.0.5 
│ │       ├── once@1.3.3 
│ │       ├─┬ readable-stream@2.1.5 
│ │       │ └── buffer-shims@1.0.0 
│ │       └── uid-number@0.0.6 
│ ├─┬ less@2.7.1
│ │ ├── image-size@0.5.0 
│ │ ├─┬ promise@7.1.1 
│ │ │ └── asap@2.0.4 
│ │ └── source-map@0.5.6 
│ ├─┬ protractor@3.3.0
│ │ └─┬ request@2.67.0
│ │   └─┬ http-signature@1.1.1
│ │     └─┬ sshpk@1.10.0
│ │       ├─┬ bcrypt-pbkdf@1.0.0 
│ │       │ └── tweetnacl@0.14.3 
│ │       ├── ecc-jsbn@0.1.1 
│ │       ├── jodid25519@1.0.2 
│ │       ├── jsbn@0.1.0 
│ │       └── tweetnacl@0.13.3 
│ └── UNMET PEER DEPENDENCY webpack@2.1.0-beta.21
├─┬ karma@0.13.22
│ └─┬ chokidar@1.6.0
│   └─┬ fsevents@1.0.14 
│     └─┬ node-pre-gyp@0.6.29 
│       ├─┬ mkdirp@0.5.1 
│       │ └── minimist@0.0.8 
│       ├─┬ nopt@3.0.6 
│       │ └── abbrev@1.0.9 
│       ├─┬ npmlog@3.1.2 
│       │ ├─┬ are-we-there-yet@1.1.2 
│       │ │ └── delegates@1.0.0 
│       │ ├── console-control-strings@1.1.0 
│       │ ├─┬ gauge@2.6.0 
│       │ │ ├── aproba@1.0.4 
│       │ │ ├── has-color@0.1.7 
│       │ │ ├── has-unicode@2.0.1 
│       │ │ ├── object-assign@4.1.0 
│       │ │ ├── signal-exit@3.0.0 
│       │ │ ├─┬ string-width@1.0.1 
│       │ │ │ ├─┬ code-point-at@1.0.0 
│       │ │ │ │ └── number-is-nan@1.0.0 
│       │ │ │ └── is-fullwidth-code-point@1.0.0 
│       │ │ ├─┬ strip-ansi@3.0.1 
│       │ │ │ └── ansi-regex@2.0.0 
│       │ │ └── wide-align@1.1.0 
│       │ └── set-blocking@2.0.0 
│       ├─┬ rc@1.1.6 
│       │ ├── deep-extend@0.4.1 
│       │ ├── ini@1.3.4 
│       │ ├── minimist@1.2.0 
│       │ └── strip-json-comments@1.0.4 
│       ├─┬ request@2.73.0 
│       │ ├── aws-sign2@0.6.0 
│       │ ├── aws4@1.4.1 
│       │ ├─┬ bl@1.1.2 
│       │ │ └── readable-stream@2.0.6 
│       │ ├── caseless@0.11.0 
│       │ ├─┬ combined-stream@1.0.5 
│       │ │ └── delayed-stream@1.0.0 
│       │ ├── extend@3.0.0 
│       │ ├── forever-agent@0.6.1 
│       │ ├─┬ form-data@1.0.0-rc4 
│       │ │ └── async@1.5.2 
│       │ ├─┬ har-validator@2.0.6 
│       │ │ ├─┬ chalk@1.1.3 
│       │ │ │ ├── ansi-styles@2.2.1 
│       │ │ │ ├── escape-string-regexp@1.0.5 
│       │ │ │ ├── has-ansi@2.0.0 
│       │ │ │ └── supports-color@2.0.0 
│       │ │ ├─┬ commander@2.9.0 
│       │ │ │ └── graceful-readlink@1.0.1 
│       │ │ ├─┬ is-my-json-valid@2.13.1 
│       │ │ │ ├── generate-function@2.0.0 
│       │ │ │ ├─┬ generate-object-property@1.2.0 
│       │ │ │ │ └── is-property@1.0.2 
│       │ │ │ ├── jsonpointer@2.0.0 
│       │ │ │ └── xtend@4.0.1 
│       │ │ └─┬ pinkie-promise@2.0.1 
│       │ │   └── pinkie@2.0.4 
│       │ ├─┬ hawk@3.1.3 
│       │ │ ├── boom@2.10.1 
│       │ │ ├── cryptiles@2.0.5 
│       │ │ ├── hoek@2.16.3 
│       │ │ └── sntp@1.0.9 
│       │ ├─┬ http-signature@1.1.1 
│       │ │ ├── assert-plus@0.2.0 
│       │ │ ├─┬ jsprim@1.3.0 
│       │ │ │ ├── extsprintf@1.0.2 
│       │ │ │ ├── json-schema@0.2.2 
│       │ │ │ └── verror@1.3.6 
│       │ │ └─┬ sshpk@1.8.3 
│       │ │   ├── asn1@0.2.3 
│       │ │   ├── assert-plus@1.0.0 
│       │ │   ├─┬ dashdash@1.14.0 
│       │ │   │ └── assert-plus@1.0.0 
│       │ │   ├── ecc-jsbn@0.1.1 
│       │ │   ├─┬ getpass@0.1.6 
│       │ │   │ └── assert-plus@1.0.0 
│       │ │   ├── jodid25519@1.0.2 
│       │ │   ├── jsbn@0.1.0 
│       │ │   └── tweetnacl@0.13.3 
│       │ ├── is-typedarray@1.0.0 
│       │ ├── isstream@0.1.2 
│       │ ├── json-stringify-safe@5.0.1 
│       │ ├─┬ mime-types@2.1.11 
│       │ │ └── mime-db@1.23.0 
│       │ ├── node-uuid@1.4.7 
│       │ ├── oauth-sign@0.8.2 
│       │ ├── qs@6.2.0 
│       │ ├── stringstream@0.0.5 
│       │ ├── tough-cookie@2.2.2 
│       │ └── tunnel-agent@0.4.3 
│       ├─┬ rimraf@2.5.3 
│       │ └─┬ glob@7.0.5 
│       │   ├── fs.realpath@1.0.0 
│       │   ├── inflight@1.0.5 
│       │   ├─┬ minimatch@3.0.2 
│       │   │ └─┬ brace-expansion@1.1.5 
│       │   │   ├── balanced-match@0.4.2 
│       │   │   └── concat-map@0.0.1 
│       │   └── path-is-absolute@1.0.0 
│       ├── semver@5.2.0 
│       ├─┬ tar@2.2.1 
│       │ ├── block-stream@0.0.9 
│       │ ├─┬ fstream@1.0.10 
│       │ │ └── graceful-fs@4.1.4 
│       │ └── inherits@2.0.1 
│       └─┬ tar-pack@3.1.4 
│         ├─┬ debug@2.2.0 
│         │ └── ms@0.7.1 
│         ├── fstream-ignore@1.0.5 
│         ├─┬ once@1.3.3 
│         │ └── wrappy@1.0.2 
│         ├─┬ readable-stream@2.1.4 
│         │ ├── buffer-shims@1.0.0 
│         │ ├── core-util-is@1.0.2 
│         │ ├── isarray@1.0.0 
│         │ ├── process-nextick-args@1.0.7 
│         │ ├── string_decoder@0.10.31 
│         │ └── util-deprecate@1.0.2 
│         └── uid-number@0.0.6 
├─┬ karma-remap-istanbul@0.2.1
│ └─┬ istanbul@0.4.5
│   └─┬ escodegen@1.8.1
│     └── source-map@0.2.0 
├── UNMET PEER DEPENDENCY rxjs@5.0.0-beta.11
└── UNMET PEER DEPENDENCY typescript@2.0.2

npm WARN @angular/compiler-cli@0.5.0 requires a peer of typescript@^1.9.0-dev but none was installed.
npm WARN @angular/compiler-cli@0.5.0 requires a peer of @angular/platform-server@^2.0.0-rc.5 but none was installed.
npm WARN @angular/core@2.0.0-rc.5 requires a peer of rxjs@5.0.0-beta.6 but none was installed.
npm WARN @angular/http@2.0.0-rc.5 requires a peer of rxjs@5.0.0-beta.6 but none was installed.
npm WARN @angular/router@3.0.0-rc.1 requires a peer of rxjs@5.0.0-beta.6 but none was installed.
npm WARN @angular/tsc-wrapped@0.2.2 requires a peer of typescript@^1.9.0-dev but none was installed.
npm WARN string-replace-loader@1.0.3 requires a peer of webpack@1.x.x || 2.x.x but none was installed.
npm WARN tsickle@0.1.6 requires a peer of typescript@^1.8.0 || ^1.9.0-dev but none was installed.
larsb-concourse:sample larsb$ ng serve
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
** NG Live Development Server is running on http://localhost:4200. **
5400ms building modules                 2ms add11420ms1258ms asset82ms emittinggHash: ed2d4bce6c4fcd99257d
Version: webpack 2.1.0-beta.21
Time: 8514ms
            Asset       Size  Chunks             Chunk Names
   main.bundle.js    2.75 MB    0, 2  [emitted]  main
 styles.bundle.js    10.2 kB    1, 2  [emitted]  styles
        inline.js    5.53 kB       2  [emitted]  inline
         main.map    3.38 MB    0, 2  [emitted]  main
       styles.map    14.1 kB    1, 2  [emitted]  styles
       inline.map    5.59 kB       2  [emitted]  inline
       index.html  476 bytes          [emitted]  
assets/.npmignore    0 bytes          [emitted]  
Child html-webpack-plugin for "index.html":
         Asset     Size  Chunks       Chunk Names
    index.html  2.81 kB       0       
webpack: bundle is now VALID.
```

```
$ ng e2e
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.

> sample@0.0.0 pree2e /Users/larsb/Documents/mycode/sample
> webdriver-manager update

[13:19:15] I/file_manager - creating folder /Users/larsb/Documents/mycode/sample/node_modules/protractor/node_modules/webdriver-manager/selenium
[13:19:15] I/downloader - selenium standalone: downloading version 2.53.1
[13:19:15] I/downloader - curl -o /Users/larsb/Documents/mycode/sample/node_modules/protractor/node_modules/webdriver-manager/selenium/selenium-server-standalone-2.53.1.jar https://selenium-release.storage.googleapis.com/2.53/selenium-server-standalone-2.53.1.jar
[13:19:15] I/downloader - chromedriver: downloading version 2.22
[13:19:15] I/downloader - curl -o /Users/larsb/Documents/mycode/sample/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.22mac32.zip https://chromedriver.storage.googleapis.com/2.22/chromedriver_mac32.zip
[13:19:16] I/update - chromedriver: unzipping /Users/larsb/Documents/mycode/sample/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.22mac32.zip
[13:19:16] I/update - chromedriver: setting permissions to 0755 for /Users/larsb/Documents/mycode/sample/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.22

> sample@0.0.0 e2e /Users/larsb/Documents/mycode/sample
> protractor "./protractor.conf.js"

[13:19:18] I/direct - Using ChromeDriver directly...
[13:19:18] I/launcher - Running 1 instances of WebDriver
Spec started

  sample App
    ✓ should fail, but does not

Executed 1 of 1 spec SUCCESS in 0.017 sec.
[13:19:21] I/launcher - 0 instance(s) of WebDriver still running
[13:19:21] I/launcher - chrome #01 passed

All end-to-end tests pass.
```

**Comment by vicb** [comment]
@hansl have you heard of such things w/ the CLI ?

**Comment by hansl** [comment]
There was a known issue with Protractor when having multiple versions of Web drivers installed in your node modules. It has been fixed with protractor 4.0.5. Make sure to install that version.

They also had changes with typings and some typings might need to be rearranged. The next CLI release will have all of that, and will be today or tomorrow.

Cheers.
